org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/StandardPreferenceManager.java

```diff
@@ -83,25 +84,40 @@ public static void initializeMavenPreferences() {
 	public void update(Preferences preferences) {
 		super.update(preferences);
 
+		boolean updateMavenProjects = false;
 		String newMavenSettings = preferences.getMavenUserSettings();
 		String oldMavenSettings = getMavenConfiguration().getUserSettingsFile();
 		if (!Objects.equals(newMavenSettings, oldMavenSettings)) {
 			try {
 				getMavenConfiguration().setUserSettingsFile(newMavenSettings);
+				updateMavenProjects = true;
 			} catch (CoreException e) {
 				JavaLanguageServerPlugin.logException("failed to set Maven settings", e);
 				preferences.setMavenUserSettings(oldMavenSettings);
 			}
 		}
-		boolean updateMavenProjects = false;
 		String newMavenGlobalSettings = preferences.getMavenGlobalSettings();
 		String oldMavenGlobalSettings = getMavenConfiguration().getGlobalSettingsFile();
 		if (!Objects.equals(newMavenGlobalSettings, oldMavenGlobalSettings)) {
 			try {
 				getMavenConfiguration().setGlobalSettingsFile(newMavenGlobalSettings);
+				updateMavenProjects = true;
 			} catch (CoreException e) {
 				JavaLanguageServerPlugin.logException("failed to set Maven global settings", e);
-				preferences.setMavenUserSettings(oldMavenSettings);
+				preferences.setMavenGlobalSettings(oldMavenGlobalSettings);
+			}
+		}
+		String newMavenLifecycleMappings = preferences.getMavenLifecycleMappings();
+		IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode(IMavenConstants.PLUGIN_ID);
+		String oldMavenLifecycleMappings = prefs.get(MavenPreferenceConstants.P_WORKSPACE_MAPPINGS_LOCATION, null);
+		if (!Objects.equals(newMavenLifecycleMappings, oldMavenLifecycleMappings)) {
+			try {
+				getMavenConfiguration().setWorkspaceLifecycleMappingMetadataFile(newMavenLifecycleMappings);
+				LifecycleMappingFactory.getWorkspaceMetadata(true);
```

Comment:
what does this line do and why is it needed?

Reply:
It reloads the lifecycle mapping. I have copied it from https://github.com/eclipse-m2e/m2e-core/blob/db42af86c7976a6131daf4cda1a10857e497bf92/org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/preferences/LifecycleMappingPreferencePage.java#L93

Reply:
Then you should consider adding a comment to clarify the purpose of this code.  Without it, other maintainers may find it hard to understand what it does from the method call alone.

Reply:
Fixed.